pkg/controller/topology.go

```diff
@@ -289,37 +291,33 @@ func getSelectedCSINode(
 		// error with the API server.
 		return nil, fmt.Errorf("error getting CSINode for selected node %q: %v", selectedNode.Name, err)
 	}
-	return selectedNodeInfo, nil
+	return selectedCSINode, nil
 }
 
 // aggregateTopologies returns all the supported topology values in the cluster that
 // match the driver's topology keys.
 func aggregateTopologies(
 	kubeClient kubernetes.Interface,
 	driverName string,
-	selectedCSINode *storage.CSINode) ([]topologyTerm, error) {
+	selectedCSINode *storage.CSINode,
+	csiNodeLister storagelisters.CSINodeLister) ([]topologyTerm, error) {
 
 	// 1. Determine topologyKeys to use for aggregation
 	var topologyKeys []string
 	if selectedCSINode == nil {
 		// Immediate binding
-
-		// TODO (#144): use informers
-		nodeInfos, err := kubeClient.StorageV1beta1().CSINodes().List(metav1.ListOptions{})
+		csiNodes, err := csiNodeLister.List(labels.Everything())
 		if err != nil {
-			// Require CSINode beta feature on K8s apiserver to be enabled.
-			// We don't want to fallback and provision in the wrong topology if there's some temporary
-			// error with the API server.
-			return nil, fmt.Errorf("error listing CSINodes: %v", err)
+			klog.Warningf("No csi nodes found")
+			return nil, nil
```

Comment:
How come it's OK to not return an error now?

Reply:
I couldn't remember any valid case for this change, I think I changed it in a wrong way, I'll change it to return error.